src/dev/src/app/datagrid-state-persistence/datagrid-state-persistence.demo.ts

```diff
@@ -1,7 +1,25 @@
 import { Component } from '@angular/core';
+import { ClrDatagridStateInterface } from '@clr/angular';
+import { of } from 'rxjs';
+import { delay } from 'rxjs/operators';
 
 @Component({
   selector: 'clr-datagrid-state-persistence-demo',
   templateUrl: './datagrid-state-persistence.demo.html',
 })
-export class DatagridStatePersistenceDemo {}
+export class DatagridStatePersistenceDemo {
+  now = new Date();
+
+  data$ = of(
+    [...Array(15).keys()].map(i => ({
+      hideableCol: 'item' + i,
+      numericCol: i,
+      dateCol: new Date(),
+      enumCol: 'Enum ' + i,
+    }))
+  ).pipe(delay(0));
+
+  refresh(state: ClrDatagridStateInterface) {
+    console.log('data refresh needed', JSON.stringify(state));
```

Comment:
intentional or forgotten here?

Reply:
intentional, because as stated in the documentation there are drawbacks and especially when you use a server filtered datagrid you need special attention on how often this method is called.